src/parser/Parser.ts

```diff
@@ -2418,17 +2466,70 @@ export class Parser {
         }
     }
 
+    private getBscTypeFromAssignment(assignment: AssignmentStatement, functionExpression: FunctionExpression): BscType {
+        try {
+            //function
+            if (isFunctionExpression(assignment.value)) {
+                let functionType = new FunctionType(assignment.value.returnType);
+                functionType.isSub = assignment.value.functionType.text === 'sub';
+                if (functionType.isSub) {
+                    functionType.returnType = new VoidType();
```

Comment:
What about sub with explicit return type?

Reply:
Good catch. I actually removed this logic because this exact logic is already handled (correctly) inside the FunctionExpression.